simplify.go

```diff
@@ -0,0 +1,88 @@
+package irutil
+
+import (
+	"math/big"
+
+	"github.com/llir/llvm/ir/constant"
+)
+
+// Simplify returns an equivalent (and potentially simplified) constant to
+// the constant expression.
+func Simplify(c constant.Constant) constant.Constant {
+	switch c := c.(type) {
+	case *constant.ExprAdd:
+		x, ok := Simplify(c.X).(*constant.Int)
+		y, ok2 := Simplify(c.Y).(*constant.Int)
+		if ok && ok2 {
+			z := constant.NewInt(x.Typ, 0)
+			z.X = (&big.Int{}).Add(x.X, y.X)
```

Comment:
Not sure if we care too much about allocations, but I think using `z.X = z.Add(x.X, y.X)` avoids one extra allocation (that of `&big.Int{}`), since `z.X` is already allocated.